internal/layers/wcow_parse.go

```diff
@@ -0,0 +1,183 @@
+//go:build windows
+// +build windows
+
+package layers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/containerd/containerd/api/types"
+
+	"github.com/Microsoft/hcsshim/internal/copyfile"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	"github.com/Microsoft/hcsshim/internal/uvmfolder"
+)
+
+// WCOW image layers is a tagging interface that all WCOW layers MUST implement. This is
+// only used so that any random struct cannot be passed as a WCOWLayers type.
+type WCOWLayers interface {
+	IsWCOWLayers()
+}
+
+// scratchLayerData contains data related to the container scratch. Scratch layer format
+// (i.e a VHD representing a scratch layer) doesn't change much across different types of
+// read-only layers (i.e WCIFS, CIMFS etc.) so this common struct is used across all other
+// layer types.
+type scratchLayerData struct {
+	// Path to the scratch layer. (In most of the cases this will be a path to the
+	// directory which contains the scratch vhd, however, in future this could be
+	// volume or a directory that is already setup for writing)
+	scratchLayerPath string
+	// Usually the container rootfs path is some volume path where the scratch VHD is
+	// mounted. If a separate `volumeMountPath` is provided this volume is also
+	// mounted there. This is mostly needed for job object containers.
+	volumeMountPath string
```

Comment:
This field appears to never be written to (only read).

Reply:
That's a good catch. This field is currently only used for JobContainers to mount the job container rootfs at another path (in cases where bind mounts are not supported). However, during the refactor I didn't correctly handle it. That's why it wasn't being written to. 
This mounting the rootfs at a different path (i.e the `volumeMountPath`) is a very job containers specific thing doesn't really need to be handled by the `layers` package. So, I have moved that code from the common WCOW mount functions into the `jobcontainer` package. Let me know what you think.